File: src/backend/yt_diffuser/types/project/project.py
from pydantic import BaseModel

from .project_layers import ProjectLayers

class Project(BaseModel):
    """
    プロジェクトの情報を保持するクラス
    """

    project_name: str
    height: int = 1024
    width: int = 1024
    version: int = 1

    layers: ProjectLayers = ProjectLayers()

File: src/frontend/src/types/project/project.js
import { ProjectLayers } from './projectLayers'

export class Project {
  constructor(data) {
    this.projectName = data?.projectName || ''
    this.width = data?.width || 1024
    this.height = data?.height || 1024
    this.layers = new ProjectLayers(data?.layers)
  }

  /**
   * レイヤーをセットする
   */
  setLayers(layers) {
    this.layers = new ProjectLayers(layers)
  }
}
